### ISSUE TITLE
Update browser_control_console.py to use asserts

### ISSUE THREAD

**Issue by mariusandreifilip** [mozilla/iris#421]

**Comment by tracywalker** [comment]
This test case doesn't work for me on Mac. It could be some Mac imagine issues like we were discussing in yesterday's team meeting.  However,  There are a few thing I can tell need to be worked on by tracing through the test case.

the initial import statement should be from iris.test_case import *

The check for the alert message uses text ocr for Mac and a pattern match for Win and Linux.  Please make it a pattern match for Mac.  Generally, we want to test things in the same way for each platform.  i personally would lean towards pattern matching, as that is the true strength of iris.  That and we're having issues with ocr reliability.

The section to open and close, min and max the console window is hard to follow.  Most of those steps can be broken out into helper functions.  There is already one for close_auxiliary_window() in helpers/general.py.

**Comment by tracywalker** [comment]
This test case still doesn't work for me on Mac.  When the case licks the Maximize button of the auxiliary window, it sends it to full screen.  You can do that , if you want to,  but you need to solve how to get the window out of full screen.  That or send the window to maximize by hovering the green button, then clicking it.  Notice the the "+" the green button turns to.   

Also, the test seems to be leaving around minimized windows, please close all windows before exiting the test case.  For some reason completion of your test case wants to force quit my terminal.

**Comment by tracywalker** [comment]
This test case still doesn't work for me on Mac.  Now it is failing to close the first console window. Instead it opens the console to full screen.  Andrei, are you running this test case on Mac?  Please have your colleagues review this test case, ensuring it works on all platforms before sending it back to me.

**Comment by mariusandreifilip** [comment]
Indeed it was a strange situation because on my Mac everything was ok and on another Firefox behaved differently.I fixed the issue and rebased the branch with the latest dev

**Comment by tracywalker** [comment]
This still doesn't work for me on Mac.  It fails attempting to find the image for minimize button of the auxiliary window.

**Comment by mariusandreifilip** [comment]
This is definitively a weird situation because we all run this test on all Os's 4 Macs 2X15'' and 2X13'' including an osx with mirror monitor.My believe is that PyautoGuy has some bugs with these images because pyautoguy doesn't know about colors and this patterns 'auxiliary_close_controls,auxiliary_minimize_controls and auxiliary_maximize' are not good.After a deeper investigation I noticed that the auxiliary_maximize is not recognized at all and the only one that works ok is the 'auxiliary_close_control' .

**Comment by tracywalker** [comment]
You are correct, pattern match is grey scale unless you set similar to .99 or exact.  I have found, with these circles in the UI, that pattern match may still fail. Perhaps try creating helpers for minimize and maximize that use click offsets from a pattern match of the window close button. That or hover the close button, then pattern match for the unique button pattern. Yes, do this, create a helper for Mac that hovers the auxiliary control button.  Then create or rework helpers for each of the buttons as they appear when in that hovered state.  You can then also create a helper for actual maximize, not full screen mode, by holding the Alt key to get the '+' in the green button before clicking it.

**Comment by mwobensmith** [comment]
Good first test case for Alin or Roxana.

**Comment by alinrus** [comment]
Refactor added in branch 'issue_421'. @tracywalker please review.

**Comment by tracywalker** [comment]
code approved
